### ISSUE TITLE
Remove orchestrator/trainer distinction

### ISSUE THREAD

**Issue by alan-cooney** [CarperAI/trlx#278]
### 🚀 The feature, motivation, and pitch

As context, it looks like the trainer classes were split up into these two classes (trainer & orchestrator), as they had become very large. 

The way this has been done is to enforce instantiating both classes, and then they add themselves to each other as a property (i.e. `trainer.orchestrator` and `orchestrator.trainer`). Whilst quite a creative solution, this is unusual and has the following disadvantages:
- Prevents type checking from working properly (as it involves circular imports)
- Hard to read (as the resulting class is still huge, and now we just have to jump between two files instead)
- Hard to debug

Put simply, the abstraction doesn't work, so I think it makes most sense to merge them back together. We can then extract functionality where it does work (i.e. smaller discrete things like the kl divergence can be extracted), which will create a far more readable and robust codebase.

**Comment by LouisCastricato** [comment]
Agreed. @jon-tow do you wanna tackle this?

**Comment by jon-tow** [comment]
The orchestrators have been merged into their respective trainers; improving on the general abstraction should be easier now.

Thanks for raising this issue Alan!
